src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -114,14 +121,39 @@ struct AttractorsViaRecurrences{DS<:DynamicalSystem, B, G, K} <: AttractorMapper
 end
 
 function AttractorsViaRecurrences(ds::DynamicalSystem, grid;
-        Dt = nothing, Δt = Dt, sparse = true, force_non_adaptive = false, kwargs...
+        Dt = nothing, Δt = Dt, sparse = true, force_non_adaptive = false, density_matrix = nothing, kwargs...
     )
-    bsn_nfo = initialize_basin_info(ds, grid, Δt, sparse)
+
+
+    if grid isa Tuple # regular or irregular
+        D = length(grid)
+        if all(t -> t isa AbstractRange, grid) # regular
+            grid_steps = SVector{D,Float64}(step.(grid))
+            grid_maxima = SVector{D,Float64}(maximum.(grid))
+            grid_minima = SVector{D,Float64}(minimum.(grid))
+            
+            finalgrid = RegularGrid(grid_steps, grid_minima, grid_maxima, grid)
+            
+        elseif any(t -> t isa AbstractVector, grid) # irregular
+            finalgrid = IrregularGrid(grid)
```

Comment:
You need to do one more check here: that all entries are also sorted. If not, throw an error. The subsequent code relies on the assumption that they are sorted.